### ISSUE TITLE
Images and tables are mangled in output

### ISSUE THREAD

**Issue by altano** [theohbrothers/ConvertOneNote2MarkDown#14]
Images and tables are not coming through correctly in the output. Backslashes in image paths are missing so I end up getting output like this:

```markdown
![](C:tempnotesNotebookmediaMyImage.png){width="12.840277777777779in" height="16.59027777777778in"}
```

But it should be `C:\temp\notes\Notebook\media\MyImage.png`

Also, tables are coming through as mangled HTML (no angle brackets) instead of markdown tables, e.g. I'm getting

```markdown
tabletheadtr class="header"thh1 id="san-diego"San Diego/h1/ththh1 id="section"
/h1/ththh1 id="palm-springs"Palm Springs/h1/ththh1 id="section-1"
/h1/ththh1 id="sacramento"Sacramento/h1/ththh1 id="section-2"
...
```

instead of

```
| San Diego | Palm Springs |
...
```

**Comment by altano** [comment]
Selecting the non-default `2: Keep '\' symbol escape` option during script run fixed images FWIW, but then I get random backslashes in the output like `Earthquake + Homeowner\'s = \$250 per month` (I understand the dollar sign is escaped because it is significant in Pandoc markdown, but why the apostrophe?)

**Comment by nixsee** [comment]
I just ran a sample note with all sorts of symbols, including apostrophe and $, as well as a simple table, and it worked fine. I used option 1 for all prompts.

Edit: oh i just added a picture and see your issue with the filepath. I'll see what i can figure out, thanks for reporting

**Comment by nixsee** [comment]
I'm very perplexed - removing the \ escape symbol seems to most definitely ruin all filepaths. Yet, I'd surely have noticed this when I first made the script - i tested with images and all sorts of other things... It might take me some time to figure this out

**Comment by nixsee** [comment]
I have found the problem - Pandoc changed the filepaths for embedded images back in October, so the code that I had to handle the paths hasn't worked since then. Bizarre that no one has reported any issues with this in the past 7 months... Maybe few people have embedded images/havent noticed it.

Anyway, I'll try to figure out something to handle this soon. Its tricky to identify the image paths only because they used to use / and now use \\, just like the escape symbols...

**Comment by altano** [comment]
Thanks @nixsee for looking into the problem. I really appreciate it.

**Comment by nixsee** [comment]
@altano It should be working now. Please let me know how it goes for you. I dont know what the problem is for your tables, however. Can you post a screenshot of what they look like in Onenote and in markdown?

**Comment by nixsee** [comment]
@altano has the updated script solved your problems?

**Comment by leojonathanoh** [comment]
@altano mangled HTML has been fixed in `v2.8.0`. As for missing backslash in paths, this might require a regular expression to only replace isolated `\` that occur in non-path like occurrences. I am in the process of converting some very large notebooks, so if i bump into this issue i will roll out a fix.

**Comment by nixsee** [comment]
I found a lot of the problems stemmed from file-path issues, but that's because I don't know how to code and isolate specific parts of the file. Additionally, pandoc changed some stuff in the way they generate filepaths, so I believe it was the most recent release that fixed something for that. Anyway, do whatever you want/need to improve all of this.

One thought is that wouldn't tables be better as markdown, not HTML? Even github just uses the pipe character to separate cells. https://github.github.com/gfm/#tables-extension-

I forget which conversion methods use this, but I think pandoc (what I decided was the best for my/most people's needs) and gfm do it.

**Comment by leojonathanoh** [comment]
> I found a lot of the problems stemmed from file-path issues, but that's because I don't know how to code and isolate specific parts of the file. Additionally, pandoc changed some stuff in the way they generate filepaths, so I believe it was the most recent release that fixed something for that. Anyway, do whatever you want/need to improve all of this.

@nixsee I really appreciate your work because your version worked out of the box and was simple enough to be understood and extended. There's always going to be cases where there are going to be bugs and edge cases but so far it's been working pretty well for some of my notebook conversions.

> One thought is that wouldn't tables be better as markdown, not HTML? Even github just uses the pipe character to separate cells. https://github.github.com/gfm/#tables-extension-
> I forget which conversion methods use this, but I think pandoc (what I decided was the best for my/most people's needs) and gfm do it.

I'm not familiar with pandoc enough but my guess is that in the scenario where a onenote page contain very complex table structures with nested images, nested drawings, nested tables, which cannot easily be converted to markdown pipe tables, then pandoc might decide to output the table in HTML. After all pandoc's philosophy is to use as little HTML as possible https://rmarkdown.rstudio.com/authoring_pandoc_markdown.html%23raw-tex#Philosophy.

**Comment by leojonathanoh** [comment]
Just wanted to update that I did bump into this issue where the `\` in image paths were stripped so I will be rolling out a fix. Reopening.

**Comment by leojonathanoh** [comment]
This issue should have been fixed with #40.

**Comment by leojonathanoh** [comment]
hey @altano, the broken image URIs in markdown should have been fixed with `v2.9.1`
